persistence/javadsl/src/main/scala/com/lightbend/lagom/internal/javadsl/persistence/AbstractPersistentEntityRegistry.scala

```diff
@@ -13,33 +13,31 @@ import akka.event.Logging
 import akka.japi.Pair
 import akka.pattern.ask
 import akka.persistence.query.scaladsl.EventsByTagQuery
-import akka.persistence.query.{ Offset => AkkaOffset }
+import akka.persistence.query.{ PersistenceQuery, Offset => AkkaOffset }
 import akka.stream.javadsl
 import akka.util.Timeout
 import akka.{ Done, NotUsed }
 import com.google.inject.Injector
 import com.lightbend.lagom.internal.persistence.cluster.GracefulLeave
 import com.lightbend.lagom.javadsl.persistence._
-
 import scala.concurrent.duration.{ FiniteDuration, _ }
 import scala.util.control.NonFatal
+import scala.compat.java8.OptionConverters._
 
 /**
  * Provides shared functionality for implementing a persistent entity registry.
  *
  * Akka persistence plugins can extend this to implement a custom registry.
  */
-abstract class AbstractPersistentEntityRegistry(system: ActorSystem, injector: Injector) extends PersistentEntityRegistry {
+class AbstractPersistentEntityRegistry(system: ActorSystem, injector: Injector) extends PersistentEntityRegistry {
 
-  /**
-   * The ID of the journal.
-   */
-  protected val journalId: String
+  protected val name: Optional[String] = Optional.empty()
+  protected val journalPluginId: String = ""
+  protected val snapshotPluginId: String = ""
```

Comment:
I'm missing an example of how these three values are being set. Is that in user code?

Reply:
Yes, user code must provide a unique name to the registry, to avoid collisions within the actor system's cluster-sharding. Journal and snapshot plugin ids must be paths in the configuration to appropriately structured plugin configurations.

A default registry is still created which behaves just like the single registry that was allowed before. It leverages existing code that reinterprets a plugin id of `""` the appropriate default plugin id.

I will try to contribute a test that uses a non-default registry.

Reply:
Maybe it would be better to have an overloaded constructor on the concrete implementations that allows you to pass in the name as a parameter (forgoing automatic dependency injection).

Config namespaces (i.e., journal and snapshot plugin IDs) could be derived from the name using a standard convention.

Then it is easy to construct them dynamically, and you could register a new singleton into Guice to manage construction and pass along the shared dependencies.